File: src/BjjEire.Seeder/EnvironmentGuard.cs
// Copyright (c) BjjEire. All rights reserved.
// Licensed under the MIT License.

namespace BjjEire.Seeder;

internal static class EnvironmentGuard
{
    internal static bool IsAllowed(bool isForce)
    {
        string environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT")
            ?? Environment.GetEnvironmentVariable("DOTNET_ENVIRONMENT")
            ?? "Development";

        if (environment.Equals("Development", StringComparison.OrdinalIgnoreCase) || isForce)
            return true;

        Console.Error.WriteLine($"ERROR: Seeder refused — environment is '{environment}'.");
        Console.Error.WriteLine("       Set ASPNETCORE_ENVIRONMENT=Development or pass --force to override.");
        return false;
    }
}

File: src/bjjeire-api/src/main/java/com/bjjeire/api/seeder/EnvironmentGuard.java
package com.bjjeire.api.seeder;

public final class EnvironmentGuard {
    private EnvironmentGuard() {}

    public static boolean isAllowed(boolean force) {
        return isAllowed(resolveEnvironment(), force);
    }

    static boolean isAllowed(String environment, boolean force) {
        if ("Development".equalsIgnoreCase(environment) || force) {
            return true;
        }

        System.err.println("ERROR: Seeder refused — environment is '" + environment + "'.");
        System.err.println("Set BJJ_ENVIRONMENT=Development (or pass --force) to override.");
        return false;
    }

    private static String resolveEnvironment() {
        String environment = firstNonBlank(
                System.getenv("BJJ_ENVIRONMENT"),
                System.getenv("ASPNETCORE_ENVIRONMENT"),
                System.getenv("DOTNET_ENVIRONMENT"));
        return environment != null ? environment : "Production";
    }

    private static String firstNonBlank(String... values) {
        for (String value : values) {
            if (value != null && !value.isBlank()) {
                return value;
            }
        }
        return null;
    }
}
